### PR TITLE
fix and test for contextual logging

### PR DESCRIPTION
**Pull request by pohly**
The logcheck finds log calls which don't support contextual logging. It now runs as part of "make test".

Fixes: https://github.com/kubernetes-sigs/dra-example-driver/issues/10

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -87,6 +87,13 @@ misspell:
 vet:
 	go vet $(MODULE)/...
 
+# Ensure that all log calls support contextual logging.
+test: logcheck
+.PHONY: logcheck
```
Comment (elezar) [inline Makefile:92]: Question: Should we run this in the GitHub actions?

**Review by elezar — COMMENTED** [review]
Thanks @pohly!

One minor question from my side.

File: Makefile (hunk 1 above)
Comment (pohly) [reply Makefile:92]: We aren't already running `test` for PRs?! We absolutely should.

I created https://github.com/kubernetes-sigs/dra-example-driver/issues/38 for E2E testing. Can you take care of doing something similar for unit testing?

Comment (elezar) [reply Makefile:92]: I have created #41

**Review by elezar — APPROVED** [review]
Thanks for this @pohly.

/lgtm

**Pull request merged**